toolkit/scripts/toolchain/cross/gcc_cross.sh

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Docs used to create this script:
+# https://preshing.com/20141119/how-to-build-a-gcc-cross-compiler/
+# https://mdeva2.home.blog/2019/07/08/building-gcc-as-a-cross-compiler-for-raspberry-pi/
+# http://jiadongsun.cc/2019/09/03/Cross_Compile_Gcc/#6-build-glibc
+
+# ====== Prerequisites ======
+# You may need to run "chmod +x <path to this script>" prior to running it to give it execute permissions
+#
+# The following packages need to be installed before running this script
+# For Ubuntu/Mariner:
+# gawk rsync bison
+
+# For Mariner, these additional packages need to be installed:
+# make patch texinfo diffutils
+# ====================
+
+
+installDir="/opt/cross"
+sysroot="${installDir}/aarch64-mariner-linux-gnu/sysroot"
+buildDir="$HOME/cross"
+scriptDir="$( cd "$( dirname "$0" )" && pwd )"
+
+sudo rm -rf ${buildDir}
+sudo rm -rf ${installDir}
+sudo rm -rf ${sysroot}
+
+mkdir -p ${buildDir}
+mkdir -p ${installDir}
+mkdir -p "${sysroot}/lib"
+
+cd ${buildDir}
+
+# Download source tarballs
+wget http://ftp.gnu.org/gnu/binutils/binutils-2.32.tar.xz
+wget https://ftp.gnu.org/gnu/gcc/gcc-9.1.0/gcc-9.1.0.tar.xz
+wget https://github.com/microsoft/CBL-Mariner-Linux-Kernel/archive/rolling-lts/mariner-2/5.15.48.1.tar.gz
+wget https://ftp.gnu.org/gnu/glibc/glibc-2.28.tar.xz
+wget https://ftp.gnu.org/gnu/mpfr/mpfr-4.0.1.tar.gz
+wget http://ftp.gnu.org/gnu/gmp/gmp-6.1.2.tar.xz
+wget https://ftp.gnu.org/gnu/mpc/mpc-1.1.0.tar.gz
```

Comment:
This looks fishy. We shouldn't be reaching upstream without validating the signatures. Also, the upstream might disappear at any time. I'd suggest relying on our blob store.